grails-app/services/org/pih/warehouse/data/ProductSupplierService.groovy

```diff
@@ -446,4 +448,82 @@ class ProductSupplierService {
         }
         return productSupplier
     }
+
+    List<Map> getExportData(ProductSupplierFilterCommand filterParams) {
```

Comment:
I still don't feel comfortable with that method. As I said in my previous comment, I would like to get rid of unnecessary :spaghetti: and make it more readable if we have a chance.
@SebastianLib please raise any issues you face, so we could figure them out together.